subsys/net/lib/sockets/socket_can.c

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2018 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define LOG_MODULE_NAME net_sock_can
+
+#define NET_LOG_LEVEL CONFIG_NET_SOCKETS_LOG_LEVEL
+
+#include <net/net_context.h>
+#include <net/socket_can.h>
+#include <device.h>
+#include <misc/printk.h>
+#include <kernel.h>
+#include <net/net_ip.h>
+#include <net/net_if.h>
+#include <net/buf.h>
+#include <net/net_pkt.h>
+#include <net/net_l2.h>
+#include <zephyr.h>
+#include <can.h>
+
+
+void socket_can_iface_init(struct net_if *iface)
+{
+	struct device *dev = net_if_get_device(iface);
+	struct socket_can_context *context = dev->driver_data;
+
+	context->iface = iface;
+}
+
+static void socket_can_tx_callback(u32_t flags)
+{
+	NET_DBG("TX CALLBACK OCCURRED flags = %x", flags);
+	if (flags != CAN_TX_OK) {
+		NET_DBG("SOCKET_CAN_TRANSMISSION FAILED ERR = %d", flags);
+	}
+}
+
+int socket_can_iface_send(struct net_if *iface, struct net_pkt *pkt)
+{
+	int ret = CAN_TX_ERR;
+	struct device *dev;
+	struct socket_can_context *context;
+	struct device *isesi_can_dev;
+	struct net_buf *buf;
+	struct can_msg *msg;
+
+	if (pkt == NULL) {
+		goto err;
+	}
+	dev = net_if_get_device(iface);
+	if (dev == NULL) {
+		goto err;
+	}
+	context = dev->driver_data;
+	if (context == NULL) {
+		goto err;
+	}
+	isesi_can_dev = context->can_dev;
+
+	buf = pkt->frags;
+	if ((buf == NULL) || (buf->data == NULL)) {
+		goto err;
+	}
+	msg = (struct can_msg *)(buf->data);
+
+	ret = can_send(isesi_can_dev, msg, K_FOREVER, socket_can_tx_callback);
+err:
+	return ret;
+}
+
+bool socket_can_check_matched_id_filter(struct net_pkt *pkt)
+{
+	struct net_if *iface = NULL;
+	struct net_context *ctx = NULL;
+	struct device *dev = NULL;
+	const struct socket_can_driver_api_t *api = NULL;
+	bool result = false;
+
+	if (pkt != NULL) {
+		ctx = net_pkt_context(pkt);
+		if (ctx != NULL) {
```

Comment:
It's a bit convoluted: only test for error, not succes. 
So it would be:

ctx = net_pkt_context(pkt);
if (ctx == NULL) {
    goto err;
}

iface = net_context_get_iface(ctx);
if (iface == NULL) {
    goto err;
}

etc...

Reply:
done